### DISCUSSION TITLE
Provide Windows friendly shortcuts

### DISCUSSION THREAD

**Discussion by florianfelsing** [AllYourBot/hostedgpt#166] [General]
I believe it would be beneficial to include shortcuts that are friendly to Windows users.

The current use of the "meta" key feels a bit awkward, since it maps to the "system" key on Windows keyboards (of which I also use one on my Ubuntu machine). It feels strange to use that key for interacting with non-system apps.

**Suggestion:**

- Support "ctrl" as an alternative input key to "meta"
- Decide if we want to block the other key, depending on what system you are on. E.g. on my Windows machine, ChatGPT won't react to "meta+shift+o", only to "ctrl+shift+o"

**Comment by keithschacht** [comment]
This is a really good point. I’m glad you use Windows. :) What does ChatGPT do for you? I listed meta+shift+o since that’s what it tells me on my Mac. Can you click the “?” on chatgpt (bottom-right) and select keyboard shortcuts and screenshot this? Here is what I see on Mac:

<img width="799" alt="image" src="https://github.com/the-dot-bot/hostedgpt/assets/35061/6af88dd0-3fa5-47e6-b4b6-d17abd37913c">

**Reply by florianfelsing** [reply]
At this very moment I am on my Ubuntu machine (which I use for development), but the shortcuts seem to be the same as when I use Windows.

![Screenshot from 2024-03-03 12-11-00](https://github.com/the-dot-bot/hostedgpt/assets/50384514/4db615cc-aa9b-48f5-b83c-22062bd1ec63)

**Comment by keithschacht** [comment]
The easiest thing to do would be to simply add Ctrl to all of our bindings. This is a task which would just take minutes. Then we’ll maintain support for Ctrl+ and Meta+. That way we don’t have to do some conditionals on each system. And then in our modal where we explain the keyboard shortcuts we could either conditionalize the modal display by system or even just show: 

```
Open/Close sidebar with     [Ctrl/Cmd] [ . ]  
```

I wish the stimulus library had something like `default_os_modifier+j` and this modifier key would default to Cmd on Mac and Ctrl on Windows. I just poked around the Stimulus source code quickly to see if there was some undocumented feature like that but I don’t see it.

**Reply by florianfelsing** [reply]
I was thinking about the same thing, and I agree.

No need to introduce additional complexity at this stage if there is no OS modifier. And a s Windows/Ubuntu user, I can always opt to simply ignore the meta key or I would have never thought of it anyway, to be honest.

**Comment by keithschacht** [comment]
Done with https://github.com/the-dot-bot/hostedgpt/pull/176
